### DISCUSSION TITLE
Public pages on account console theme

### DISCUSSION THREAD

**Discussion by xoxys** [keycloak/keycloak#9014] [Q&A] [answered]
Hi,

is it possible to embed a public page in an account console custom theme? We would like to provide a help page for users to set up 2FA that can be reached without login. Something like this: 

![image](https://user-images.githubusercontent.com/3391958/144869061-756e2cab-30bb-45d5-92f8-7aaf6dfed0f0.png)

If possible, we would like to avoid hosting the page on another web server and serve the page from the theme itself.

**Comment by ssilvert** [comment]
Yes, it's quite easy to add pages like that.

Mostly it's just a matter of adding an entry in content.json.

See the "Keycloak Man" example for more details about customizing the
account UI.
https://github.com/keycloak/keycloak-quickstarts/tree/main/extend-account-console

Stan

**Reply by xoxys** [reply]
Thanks. I've tried that already, maybe I did it wrong, but the pages were not accessible **without** an active login or am I wrong?

**Comment by ssilvert** [comment, marked as answer]
Sorry.  I misunderstood.

The index.ftl page does not require login.  So I would just embed your page
into that.  You could just show or hide your "Getting Started" info inside
the content area.

Stan

**Reply by xoxys** [reply]
Ok got it. I don't think it looks great to embed a "complex" setup guide into the index front page, but I'll think about it again.

**Comment by ssilvert** [comment]
I agree.  We should allow you to add pages to content.json that don't
require login.  I'll keep that in mind as a new feature when we do a major
refactoring of the UI next year.
